Ekubo wETH-ETH liquidity pool.ipynb

```diff
@@ -0,0 +1,1405 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "ef26a9ee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# To store valid events\n",
+    "valid_events_positions = []\n",
+    "valid_events_swap = []\n",
+    "\n",
+    "# Processing positions updated events\n",
+    "for event_no in range(len(events_response_positions_updated.events)):\n",
+    "    event_data = events_response_positions_updated.events[event_no].data\n",
+    "    \n",
+    "    # Check for the first condition\n",
+    "    if (hex(event_data[2]) == \"0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7\" and \\\n",
+    "        hex(event_data[1]) == \"0x42b8f0484674ca266ac5d08e4ac6a3fe65bd3129795def2dca5c34ecc5f96d2\" and \\\n",
+    "        event_data[3] == pool_fee_touse and \\\n",
+    "        event_data[4] == tick_spacing_touse):\n",
+    "        valid_events_positions.append(event_data)  # Save the valid event data\n",
+    "    \n",
+    "    # Check for invalid order condition\n",
+    "    elif (hex(event_data[1]) == \"0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7\" and \\\n",
+    "          hex(event_data[2]) == \"0x42b8f0484674ca266ac5d08e4ac6a3fe65bd3129795def2dca5c34ecc5f96d2\"):\n",
+    "        \n",
+    "        raise Exception(\"invalid order\")\n",
+    "\n",
+    "# Processing swap events\n",
+    "for event_no_swap in range(len(combined_events_response_swap)):\n",
+    "    event_data = combined_events_response_swap[event_no_swap].data\n",
+    "    \n",
+    "    # Check for the first condition\n",
+    "    if (hex(event_data[2]) == \"0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7\" and \\\n",
+    "        hex(event_data[1]) == \"0x42b8f0484674ca266ac5d08e4ac6a3fe65bd3129795def2dca5c34ecc5f96d2\" and \\\n",
+    "        event_data[3] == pool_fee_touse and \\\n",
+    "        event_data[4] == tick_spacing_touse):\n",
+    "        valid_events_swap.append(event_data)  # Save the valid event data\n",
+    "    \n",
+    "    # Check for invalid order condition\n",
+    "    elif (hex(event_data[1]) == \"0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7\" and \\\n",
+    "          hex(event_data[2]) == \"0x42b8f0484674ca266ac5d08e4ac6a3fe65bd3129795def2dca5c34ecc5f96d2\"):\n",
+    "        raise Exception(\"invalid order\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "eee36116",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "576\n",
+      "20248\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(len(valid_events_positions))\n",
+    "print(len(valid_events_swap))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "96078130",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def bool_to_sign(x):\n",
```

Comment:
Functions like this are generic and since are meant to be utility, they are to be added in utils file.